packages/react/slider/src/Slider.tsx

```diff
@@ -486,28 +488,44 @@ SliderRange.displayName = RANGE_NAME;
 const THUMB_NAME = 'SliderThumb';
 const THUMB_DEFAULT_TAG = 'span';
 
-type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'value' | 'index'>;
+type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'index'>;
 type SliderThumbPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof SliderThumbImpl>,
   SliderThumbOwnProps
 >;
 
 const SliderThumb = React.forwardRef((props, forwardedRef) => {
-  const { ref: collectionRef, index } = useSliderCollectionItem();
-  const ref = useComposedRefs(forwardedRef, collectionRef);
-  const context = useSliderContext(THUMB_NAME);
-  const value = context.values[index];
-  return value !== undefined ? (
-    <SliderThumbImpl {...props} ref={ref} index={index} value={value} />
+  const { getItems } = useCollection();
+  const [thumb, setThumb] = React.useState<React.ElementRef<typeof SliderThumbImpl> | null>(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setThumb(node));
+  const isInitialRenderRef = React.useRef(true);
+  const index = React.useMemo(
+    () => (thumb ? getItems().findIndex((item) => item.ref.current === thumb) : -1),
+    [getItems, thumb]
+  );
+
+  React.useEffect(() => {
+    isInitialRenderRef.current = false;
+  }, []);
+
+  /**
+   * We hide thumbs on the first render while we work out the index, otherwise SSR will
+   * render them in the wrong position before they snap into the correct position during
+   * hydration which would be visually jarring.
+   */
+  return isInitialRenderRef.current || index !== -1 ? (
```

Comment:
If we're checking for the initial render case here, is the `index !== -1` just a safety to not render the whole thing if for some reason that thumb wasn't tracked? I can't think of any reason why this check would be necessary right now.

Reply:
It's necessary for this case:

```jsx
<Slider.Root value={[20]}>
	...
	<Slider.Thumb />
	<Slider.Thumb />
</Slider.Root>
```

Notice how there are two thumbs but only one value? The second thumb won't render. We handled this case before with the `value !== undefined` check but perhaps it's okay to let them break it and remove redundant JSX instead?

Reply:
Yeah that’s probably fine too seeing as it’s just a user error that they would figure out pretty easily.

Reply:
I can create a PR for that then :+1:

Reply:
Oh actually, I just thought about this some more... the `SliderImpl` requires an `index`. So, it would be wrong for me to pass an index that doesn't exist. 

Perhaps it would be less confusing if I made the index fallback to `undefined` and did a `index !== undefined` check here instead?